dspace-api/src/main/java/org/dspace/external/provider/impl/SHERPAv2PublisherDataProvider.java

```diff
@@ -0,0 +1,361 @@
+        int timeout = 5000;
+        method.setConfig(RequestConfig.custom()
+            .setConnectionRequestTimeout(timeout)
+            .setConnectTimeout(timeout)
+            .setSocketTimeout(timeout)
+            .build());
+        return new HttpGet(uriBuilder.build());
+    }
+
+    private ExternalDataObject constructExternalDataObjectFromSherpaPublisher(SHERPAPublisher sherpaPublisher) {
+        ExternalDataObject externalDataObject = new ExternalDataObject();
+        externalDataObject.setSource(sourceIdentifier);
+
+        // Set publisher name
+        if (StringUtils.isNotBlank(sherpaPublisher.getName())) {
+            externalDataObject.addMetadata(new MetadataValueDTO(
+                "dc", "title", null, null, sherpaPublisher.getName()));
+            externalDataObject.setDisplayValue(sherpaPublisher.getName());
+            externalDataObject.setValue(sherpaPublisher.getName());
+        }
+        // Set publisher ID
+        if (StringUtils.isNotBlank(sherpaPublisher.getIdentifier())) {
+            externalDataObject.setId(sherpaPublisher.getIdentifier());
+            externalDataObject.addMetadata(new MetadataValueDTO(
+                "dc", "identifier", "sherpaPublisher", null,
+                sherpaPublisher.getIdentifier()));
+        }
+
+        // Set home URL
+        if (StringUtils.isNotBlank(sherpaPublisher.getUri())) {
+            externalDataObject.addMetadata(new MetadataValueDTO(
+                "dc", "identifier", "other", null, sherpaPublisher.getUri()));
+        }
+
+        return externalDataObject;
+    }
+
+    @Override
+    public boolean supports(String source) {
+        return StringUtils.equalsIgnoreCase(sourceIdentifier, source);
+    }
+
+    /**
+     * Get number of results returned from a SHERPA v2 publication search
+     * @param query The query to be search on and give the total amount of results
+     * @return int representing number of publisher results
+     */
+    @Override
+    public int getNumberOfResults(String query) {
+        HttpGet get = null;
+        try {
+            get = constructHttpGet(query);
+            HttpClient hc = new DefaultHttpClient();
+            HttpResponse response = hc.execute(get);
+            if (response.getStatusLine().getStatusCode() == 200) {
+                // Read response and parse as SHERPAResponse object
+                InputStream content = null;
+                try {
+                    content = response.getEntity().getContent();
+                    SHERPAPublisherResponse sherpaResponse =
+                        new SHERPAPublisherResponse(content, SHERPAPublisherResponse.SHERPAFormat.JSON);
+                    if (CollectionUtils.isNotEmpty(sherpaResponse.getPublishers())) {
+                        return sherpaResponse.getPublishers().size();
+                    }
+                } catch (IOException e) {
+                    log.error("Error reading input stream for SHERPAResponse: " + e.getMessage());
```

Comment:
I have now rethrown this exception in 3ef47dac - this one I can do, as it is nested within a higher try/catch block.
However, note that for the main catch in the method:
This class and its methods implement ExternalDataService interface (via extending ExternalDataServiceImpl). The abstract methods don't have any throw statements in its signatures. I would be hesitant to include a refactor of this interface and abstract impl in this PR as it would have consequences for all other providers.